File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    index = 0
    hambergers = []
    
    for i in ingredient:
        hambergers.append(i)
        if len(hambergers) >= 4:
            if hambergers[-4:] == [1, 2, 3, 1]:
                answer += 1
                for _ in range(4):
                    hambergers.pop()
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    const hamburger = []
    var answer = 0;
    for (let i = 0; i < ingredient.length; i++) {
        hamburger.push(ingredient[i])
        
        if (hamburger.length >= 4) {
            if (hamburger.slice(hamburger.length - 4).join("") === "1231") {
                hamburger.splice(hamburger.length - 4, 4)
                answer++
            }
        }
    }
    
    return answer;
}